blog/2019-10-02-MigratingOptaWebEmployeeRosteringToSpringBoot.adoc

```diff
@@ -0,0 +1,195 @@
+
+== REST API
+
+Now let's actually interact with the backend application.
+An easy way to test the REST API is to use the Swagger framework, which OptaWeb Employee Rostering supports.
+
+As an example, here are the `Tenant` REST methods shown in the Swagger user interface:
+
+image::tenantMethods.png[]
+
+As we can see, each method has documentation, the relative path URL, and its corresponding HTTP request.
+The Swagger UI makes it very easy to test any method in the backend.
+Let's say we wanted a list of all the Tenants in the application.
+We start by selecting the corresponding method to display the method details:
+
+image::getTenantsMethod.png[]
+
+Here, the `Example Value` field shows a sample value of a Tenant list, serialized to JSON format.
+We see that each Tenant contains the `id`, `name`, and `version` fields, which are saved in the database whenever a Tenant is added.
+Also note the `Try it out` button in the top right corner.
+We can use it to send HTTP requests with ease for testing purposes.
+Let's try it out:
+
+image::executeGetTenants.png[]
+
+An `Execute` button will appear, and clicking on it sends a GET request that gets all the saved Tenants in JSON format.
+Notice the GET request is sent to the `Request URL`, and the response is found in the `Response body`.
+The server returns the `200` code, which means the request was handled with no issues.
+
+Nice, we just sent our first HTTP request!
+But the Employee Rostering application does much more than just getting entities.
+There are methods for creating, deleting, and updating various entities (remember the Domain Model above?), which the application uses to modify the database, as well as methods that interact with the OptaPlanner API.
+
+Let's try a more interesting method:
+
+image::solveMethod.png[]
+
+This method is the core of the OptaWeb Employee Rostering application.
+It starts the OptaPlanner solver that assigns work shifts to employees for a specific Tenant.
+The solver uses heuristics and algorithms to find the most optimal solution under time and resource constraints.
+
+Let's try it out for the Tenant with `id = 1`:
+
+image::executeSolverMethod.png[]
+
+After clicking `Execute`, the solver starts evaluating possible solutions for arranging shifts, and finds the shift roster with the best score.
+The score is calculated using the Drools business constraint rules that we configured in the application.
+As the algorithm finds new solutions, the engine evaluates the score of each solution and compares it with the score of the current best solution.
+If the score is higher, the new solution becomes the new best solution.
+We can see this process in the messages the logger produces during solving:
+
+image::solverLog.png[]
+
+Congratulations!
+We've just found an optimal employee shift arrangement given the time and business constraints for this Tenant.
+Notice how the medium score at the start of solving (-660) was improved to (0) at the end.
```

Comment:
You may want to translate this to a language used by humans. What does it mean, for the score to change? Give an example, such as "where we needed 5 employees before, we now only need 3".

Reply:
> > Thank you for the review @triceo ! Your general comment is very insightful, I did struggle to figure out what the main idea of the blog is. I focus on new features of the refactored app rather than the refactoring itself. Since this blog post will likely be a two part series (@Christopher-Chianelli writing about the frontend and I'm writing about the backend), what about `What's New in OptaWeb Employee Rostering? (part 2) - Migrating Backend Application to Spring Boot`?
> 
> Thinking about it again, I think we could use something a little more buzzword-y. `What's New in OptaWeb Employee Rostering, part 2: Makind the backend cloud-ready`? Perhaps others will have better ideas.

Haha I'm not the best person for buzzword generation, but your suggestion sure beats mine.